dali/operators/image/resize/resampling_attr.cc

```diff
@@ -0,0 +1,122 @@
+  .AddOptionalArg("interp_type",
+      R"code(Type of interpolation used. Use `min_filter` and `mag_filter` to specify
+different filtering for downscaling and upscaling.)code",
+      DALI_INTERP_LINEAR, true)
+  .AddOptionalArg("mag_filter", "Filter used when scaling up",
+      DALI_INTERP_LINEAR, true)
+  .AddOptionalArg("min_filter", "Filter used when scaling down",
+      DALI_INTERP_LINEAR, true)
+  .AddOptionalArg<DALIDataType>("dtype", "Output data type; must be same as input type or `float`. "
+     "If not set, input type is used.", nullptr)
+  .AddOptionalArg("temp_buffer_hint",
+      "Initial size, in bytes, of a temporary buffer for resampling.\n"
+      "Ingored for CPU variant.\n",
+      0)
+  .AddOptionalArg("minibatch_size", "Maximum number of images processed in a single kernel call",
+      32);
+
+
+using namespace kernels;  // NOLINT
+
+inline ResamplingFilterType interp2resample(DALIInterpType interp) {
+#define DALI_MAP_INTERP_TO_RESAMPLE(interp, resample) case DALI_INTERP_##interp:\
+  return ResamplingFilterType::resample;
+
+  switch (interp) {
+    DALI_MAP_INTERP_TO_RESAMPLE(NN, Nearest);
+    DALI_MAP_INTERP_TO_RESAMPLE(LINEAR, Linear);
+    DALI_MAP_INTERP_TO_RESAMPLE(CUBIC, Cubic);
+    DALI_MAP_INTERP_TO_RESAMPLE(LANCZOS3, Lanczos3);
+    DALI_MAP_INTERP_TO_RESAMPLE(GAUSSIAN, Gaussian);
+    DALI_MAP_INTERP_TO_RESAMPLE(TRIANGULAR, Triangular);
+  default:
+    DALI_FAIL("Unknown interpolation type");
+  }
+#undef DALI_MAP_INTERP_TO_RESAMPLE
+}
+
+void ResamplingFilterAttr::PrepareFilterParams(
+      const OpSpec &spec, const ArgumentWorkspace &ws, int num_samples) {
+  GetPerSampleArgument(interp_type_arg_, "interp_type", spec, ws, num_samples);
+  GetPerSampleArgument(min_arg_, "min_filter", spec, ws, num_samples);
+  GetPerSampleArgument(mag_arg_, "mag_filter", spec, ws, num_samples);
+  if (!spec.TryGetArgument(dtype_arg_, "dtype"))
+    dtype_arg_ = DALI_NO_TYPE;
+  bool has_interp = spec.ArgumentDefined("interp_type");
+  bool has_min = spec.ArgumentDefined("min_filter");
+  bool has_mag = spec.ArgumentDefined("mag_filter");
+
+  min_filter_.resize(num_samples, ResamplingFilterType::Triangular);
+  mag_filter_.resize(num_samples, ResamplingFilterType::Linear);
+
+  auto convert = [](auto &filter_types, auto &interp_types) {
+    for (int i = 0, n = filter_types.size(); i < n; i++)
+      filter_types[i] = interp2resample(interp_types[i]);
+  };
+
+  if (has_min)
+    convert(min_filter_, min_arg_);
+  else if (has_interp)
+    convert(min_filter_, interp_type_arg_);
+
+  if (has_mag)
+    convert(mag_filter_, mag_arg_);
+  else if (has_interp)
+    convert(mag_filter_, interp_type_arg_);
+
+  spec.GetArgument<int64_t>("temp_buffer_hint");
+}
+
+void ResamplingFilterAttr::GetResamplingParams(
+      span<kernels::ResamplingParams> resample_params,
+      span<const ResizeParams> resize_params) const {
+  int p = 0;
+  for (int i = 0; i < resize_params.size(); i++) {
```

Comment:
```suggestion
  for (int i = 0, p = 0; i < resize_params.size(); i++) {
```
nitpick: To align with the code below